### ISSUE TITLE
Response type is hardcoded to ResponseType.Stream in the DioClientDelegate

### ISSUE THREAD

**Issue by MagnusJohansson** [dart-ogurets/dart-openapi-maven#65]
When trying to use the DioClientDelegate (to utilize the [dio_cache_interceptor](https://pub.dev/packages/dio_cache_interceptor) package;

```

    // Global options
    final options = CacheOptions(
      // A default store is required for interceptor.
      store: MemCacheStore(),
      // Optional. Returns a cached response on error but for statuses 401 & 403.
      hitCacheOnErrorExcept: [401, 403],
      // Optional. Overrides any HTTP directive to delete entry past this duration.
      maxStale: const Duration(days: 7),
    );

    final dio = Dio()..interceptors.add(DioCacheInterceptor(options: options));

    _apiClient =
        ApiClient(basePath: apiUrl, apiClientDelegate: DioClientDelegate(dio));
```
The dio_cache_interceptor crashes with `Unhandled Exception: type 'ResponseBody' is not a subtype of type 'Stream<List<int>>' in type cast
`:
![image](https://user-images.githubusercontent.com/674909/126903306-5638b8aa-c3e2-4c40-9db5-3e5b5857dc88.png)
It seems the reason is that the ResponseType passed in is Stream, even though the Reponse is of content type json:

Even though I explicitly set the ResponseType to json:
![image](https://user-images.githubusercontent.com/674909/126903352-a83edcf1-51d3-4104-9220-09a2cf2c81c4.png)

It seems that the invokeApi method is hard coded to pass ResponseType.stream:

![image](https://user-images.githubusercontent.com/674909/126903247-082396ad-525c-4296-b024-7e10a671bbf2.png)

**Comment by rvowles** [comment]
Yes, thats because everything relies on it being a stream. Are you able to intercept it on the way out and change the Options header when you need and change it back? Its a bit edge case for me specifically to deal with, if you have ideas on being able to resolve it without breaking anything I'm absolutely happy to accomodate?

**Comment by rvowles** [comment]
And the DioClientDelegate is replaceable, you will see it just implements ApiClientDelegate - so you can create your own delegate and see how you get on if you like? There used to be a Web and a IO based one before Dio supported it directly (and before we supported Dio) which is why it exists, but you could definitely use it here to replace the DioClientDelegate.

**Comment by rvowles** [comment]
Were you able to get this going the way you wanted?

**Comment by MagnusJohansson** [comment]
@rvowles No, I never managed to get dio_cache_interceptor working with this generator. Too bad. If you have a sample it would be great.

**Comment by Maczuga** [comment]
Might be quite late, but I have managed to combine the OpenApi generated code with dio_cache_interceptor:

Gist: https://gist.github.com/Maczuga/255e822a09f8d3dd8284096e5cda3003

The key is to hook the new delegate before using the client (I have just created my own client that I use). I am also using the `options.responseType = ResponseType.plain` because it's seems to be the easiest way to convert the plain json into a stream.

Server-side all I have to do is to emit the Expires header and caching works.

Edit: Oh, forgot to mention - make sure that your OpenAPI json/yaml definition file includes 304 as a response type, instead your response will be treaten as invalid:
![image](https://user-images.githubusercontent.com/544349/131516206-277b9034-e0c4-41c7-8b43-e674f5abedff.png)
